File: Doctor.API/DigitalTwin.Application/Enums/VitalSignType.cs
namespace DigitalTwin.Application.Enums;

/// <summary>
/// Defines vital-sign types exposed by the application layer.
/// </summary>
public enum VitalSignType
{
    /// <summary>
    /// Represents heart rate samples.
    /// </summary>
    HeartRate = 0,

    /// <summary>
    /// Represents blood oxygen saturation samples.
    /// </summary>
    SpO2 = 1,

    /// <summary>
    /// Represents step count samples.
    /// </summary>
    Steps = 2,

    /// <summary>
    /// Represents calorie samples.
    /// </summary>
    Calories = 3,

    /// <summary>
    /// Represents active energy samples.
    /// </summary>
    ActiveEnergy = 4,

    /// <summary>
    /// Represents exercise minute samples.
    /// </summary>
    ExerciseMinutes = 5,

    /// <summary>
    /// Represents stand hour samples.
    /// </summary>
    StandHours = 6
}

File: SwiftUIApp/DigitalTwinApp/Domain/Models/VitalSignType.swift
import Foundation

enum VitalSignType: Int, Codable, CaseIterable {
    case heartRate = 0
    case spO2 = 1
    case steps = 2
    case calories = 3
    case activeEnergy = 4
    case exerciseMinutes = 5
    case standHours = 6

    var displayName: String {
        switch self {
        case .heartRate: return "Heart Rate"
        case .spO2: return "SpO2"
        case .steps: return "Steps"
        case .calories: return "Calories"
        case .activeEnergy: return "Active Energy"
        case .exerciseMinutes: return "Exercise Minutes"
        case .standHours: return "Stand Hours"
        }
    }

    var unit: String {
        switch self {
        case .heartRate: return "bpm"
        case .spO2: return "%"
        case .steps: return "steps"
        case .calories: return "kcal"
        case .activeEnergy: return "kcal"
        case .exerciseMinutes: return "min"
        case .standHours: return "hrs"
        }
    }
}

